R-packages/evalcast/R/get_predictions.R

```diff
@@ -0,0 +1,119 @@
+#' Get predictions
+#'
+#' For each of the provided forecast dates, runs a forecaster using the data
+#' that would have been available as of that given forecast date. Returns a list
+#' of "predictions cards."  A prediction card is a data frame giving the
+#' forecast distributions of a given forecaster for a given forecast task.  A
+#' forecast task is specified by the forecast date, ahead, response, incidence
+#' period, and geo_type (e.g., 1-epiweek ahead death forecasting at the state
+#' level with predictions made using the information as of Sept. 14).
+#'
+#' A predictions card has two columns:
+#' \enumerate{
+#' \item location - the FIPS code of the location.  For counties, this is the
+#' same as the geo_value used in the API.  However, for states, the location
+#' and geo_value are different because the API uses state abbreviations instead
+#' of FIPS for geo_value.
+#' \item forecast_distribution - this is a list column... each element itself
+#' contains a tibble with the covidhub quantiles.}
+#' A predictions card has attributes that specify the exact forecasting
+#' task that was being carried out and the name of the forecaster.
+#'
+#' @param forecaster a function that outputs a tibble with columns...
+#' @param name_of_forecaster the name of forecaster
+#' @param forecast_dates a vector of class Date on which forecasts will be made.
+#'   e.g. \code{c(lubridate::ymd("2020-09-07"), lubridate::ymd("2020-09-14"))}
+#' @template signals-template
+#' @template incidence_period-template
+#' @template ahead-template
+#' @template geo_type-template
+#' @param geo_values see \link[covidcast]{covidcast_signal} for a description
+#'   of this parameter.
+#' @return a list of predictions cards
+#' @importFrom assertthat assert_that
+#' @export
+get_predictions <- function(forecaster,
+                            name_of_forecaster,
+                            signals,
+                            forecast_dates,
+                            incidence_period,
+                            ahead,
+                            geo_type,
```

Comment:
@jacobbien Wondering if there's a reason to have repeated arguments here, since it seems like `signals`, `incidence_period`, `ahead`, and `geo_type` would all typically be known by the supplied `forecaster` itself.  Could these just not be read off from the `forecaster`?  For example, they could be saved as attributes of `forecaster`.  

As it is now, it looks like with your `baseline_forecaster()` example, I have to specify `signals`, `incidence_period`, `ahead`, and `geo_type` twice: once in the call to `baseline_forecaster()`, and once in the call to `get_predictions()`.  Not a big deal, just pointing it out to see if we can simplify.

Reply:
Ahhh never mind.  I think I misunderstood the workflow (please do send me that notebook if you get a chance!).  I understand it now: `get_predictions()` simply calls `forecaster` and passes on these arguments to it.  My bad!  You can resolve this convo.